Client/Frontend/Browser/TabDisplayManager.swift

```diff
@@ -1102,16 +1101,21 @@ extension TabDisplayManager: TabManagerDelegate {
         /// collectionView is not visible the dataSource section/items differs from the actions to be perform
         /// which causes the crash
         collectionView.numberOfItems(inSection: 0)
-        collectionView.performBatchUpdates({
-            operation()
-        }, completion: { [weak self] (done) in
-            self?.performingChainedOperations = false
-            self?.tabDisplayCompletionDelegate?.completedAnimation(for: type)
-            self?.performChainedOperations()
-        })
+        /// Add temporary hack to delay the `collectionView.performBatchUpdates`
+        /// to fix crash caused by race condition where the dataSource hasn't finish loading causing Internal inconsistency crash
+        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(200)) {
```

Comment:
This is a hack fix but the compromise is that this code will be refactored in the near future so the hack makes sense as temporary instead of refactoring before the Tabs refactoring

Reply:
Hmm, there has to be a better way then adding a UI delay. I recall doing some work in this area.
Is the issue here that we get the crash only on deletion of last inactive cell?

Reply:
yes but also fixes other crashes when the Tabs refactor flag is On